src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5922,12 +5938,22 @@ private void ParseTypeArgumentList(out SyntaxToken open, SeparatedSyntaxListBuil
             // remaining types & commas
             while (true)
             {
-                if (this.CurrentToken.Kind == SyntaxKind.GreaterThanToken)
+                // We prefer parsing as if there was a missing > over parsing as a tuple type for error recovery
+                if (tokenKindBreaksArgumentList(this.CurrentToken.Kind) || this.CurrentToken.Kind is SyntaxKind.GreaterThanToken)
                 {
                     break;
                 }
```

Comment:
```suggestion
                if (this.CurrentToken.Kind is SyntaxKind.GreaterThanToken)
                    break;

                // Docs
                if (tokenKindBreaksArgumentList(this.CurrentToken.Kind))
                    break;
```

we don't need to merge blocks just because they both break.  Having separate checks makes it easier to understand (and document) what is going on.